hathor/consensus/poa/poa_block_producer.py

```diff
@@ -70,14 +66,9 @@ def __init__(self, *, settings: HathorSettings, reactor: ReactorProtocol, poa_si
         self._manager: HathorManager | None = None
         self._poa_signer = poa_signer
         self._last_seen_best_block: Block | None = None
-
-        self._started_producing = False
-        self._start_producing_lc = LoopingCall(self._start_producing)
-        self._start_producing_lc.clock = self._reactor
-
-        self._schedule_block_lc = LoopingCall(self._schedule_block)
-        self._schedule_block_lc.clock = self._reactor
         self._delayed_call: IDelayedCall | None = None
+        self._start_producing_lc: LoopingCall = LoopingCall(self._safe_start_producing)
```

Comment:
I still don't understand why we need this `LoopingCall`. Anyway, I'm ok to get this PR merged and get this changed later because it has no impact on mainnet and testnet.

Reply:
Using either a `LoopingCall` or a recursive `callLater`, separate from the main `callLater` logic (triggered reactively via pubsub) were the only good ways I found to implement this while covering the following requirements:

- A single full node should be able to produce blocks by itself.
- If block production throws during the "start producing" phase, production should be tried again. However, if it fails after a call via pubsub, it should **not** be retried.
- We have to consider that the call to `can_start_mining()` may throw an exception.

I created an issue to review this later: https://github.com/HathorNetwork/hathor-core/issues/1097